src/Microsoft.AspNetCore.Hosting.Abstractions/IWebHostBuilder.cs

```diff
@@ -82,10 +88,16 @@ public interface IWebHostBuilder
         IWebHostBuilder UseSetting(string key, string value);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete and will be removed in a future version. The recommended alternative is <see cref="UseLoggerFactory(Func{WebHostBuilderContext, ILoggerFactory})"/>.
+        /// </para>
+        /// <para>
         /// Specify the <see cref="ILoggerFactory"/> to be used by the web host.
+        /// </para>
         /// </summary>
         /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to be used.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.AspNetCore.Hosting.IwebHostBuilder.UseLoggerFactory(Func<WebHostBuilderContext, ILoggerFactory> createLoggerFactory).")]
         IWebHostBuilder UseLoggerFactory(ILoggerFactory loggerFactory);
```

Comment:
I believe you are right. @glennc when I originally filed the issue I remember we said we wanted to deprecate both but I don't see why we want to force anyone to use the overload with the context if it's not needed.